x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedManager.java

```diff
@@ -198,6 +198,41 @@ public void getDatafeeds(GetDatafeedsAction.Request request, ClusterState state,
         ));
     }
 
+    public void getDatafeedsByJobIds(Set<String> jobIds, ClusterState state, ActionListener<Map<String, DatafeedConfig.Builder>> listener) {
+        datafeedConfigProvider.findDatafeedsByJobIds(jobIds, ActionListener.wrap(
+            datafeeds -> {
+                Map<String, DatafeedConfig.Builder> response = new HashMap<>(datafeeds);
+                Map<String, DatafeedConfig> fromState = MlMetadata.getMlMetadata(state).getDatafeedsByJobIds(jobIds);
```

Comment:
Are all the datafeeds stored in cluster state? Or is this code only to serve the legacy datafeeds (created before ML configs were moved from cluster state to index)?

Reply:
This is legacy only. Datafeeds are supposed to be fully migrated into the configuration index.